swarm/network/stream/streamer_test.go

```diff
@@ -1293,7 +1293,9 @@ func TestGetSubscriptionsRPC(t *testing.T) {
 			t.Fatal("Context timed out")
 		}
 
+		lock.RLock()
```

Comment:
This lock belongs to the `expectedMsgCount`. What do you think about introducing a helper type? So when I'm reading the test I don't have to look around what and why do we lock.

Something like:
```
diff --git a/swarm/network/stream/streamer_test.go b/swarm/network/stream/streamer_test.go
index 185081dee..57a3b7b85 100644
--- a/swarm/network/stream/streamer_test.go
+++ b/swarm/network/stream/streamer_test.go
@@ -1177,6 +1177,7 @@ starts the simulation, waits for SyncUpdateDelay in order to kick off
 stream registration, then tests that there are subscriptions.
 */
 func TestGetSubscriptionsRPC(t *testing.T) {
+
 	// arbitrarily set to 4
 	nodeCount := 4
 	// run with more nodes if `longrunning` flag is set
@@ -1188,19 +1189,16 @@ func TestGetSubscriptionsRPC(t *testing.T) {
 	// holds the msg code for SubscribeMsg
 	var subscribeMsgCode uint64
 	var ok bool
-	var expectedMsgCount = 0
+	var expectedMsgCount msgCount
 
 	// this channel signalizes that the expected amount of subscriptiosn is done
 	allSubscriptionsDone := make(chan struct{})
-	lock := sync.RWMutex{}
 	// after the test, we need to reset the subscriptionFunc to the default
 	defer func() { subscriptionFunc = doRequestSubscription }()
 
 	// we use this subscriptionFunc for this test: just increases count and calls the actual subscription
 	subscriptionFunc = func(r *Registry, p *network.Peer, bin uint8, subs map[enode.ID]map[Stream]struct{}) bool {
-		lock.Lock()
-		expectedMsgCount++
-		lock.Unlock()
+		expectedMsgCount.Inc()
 		doRequestSubscription(r, p, bin, subs)
 		return true
 	}
@@ -1293,9 +1291,7 @@ func TestGetSubscriptionsRPC(t *testing.T) {
 			return errors.New("Context timed out")
 		}
 
-		lock.RLock()
-		log.Debug("Expected message count: ", "expectedMsgCount", expectedMsgCount)
-		lock.RUnlock()
+		log.Debug("Expected message count: ", "expectedMsgCount", expectedMsgCount.Count())
 		//now iterate again, this time we call each node via RPC to get its subscriptions
 		realCount := 0
 		for _, node := range nodes {
@@ -1326,9 +1322,7 @@ func TestGetSubscriptionsRPC(t *testing.T) {
 			}
 		}
 		// every node is mutually subscribed to each other, so the actual count is half of it
-		lock.RLock()
-		emc := expectedMsgCount
-		lock.RUnlock()
+		emc := expectedMsgCount.Count()
 		if realCount/2 != emc {
 			return fmt.Errorf("Real subscriptions and expected amount don't match; real: %d, expected: %d", realCount/2, emc)
 		}
@@ -1338,3 +1332,20 @@ func TestGetSubscriptionsRPC(t *testing.T) {
 		t.Fatal(result.Error)
 	}
 }
+
+type msgCount struct {
+	lock  sync.RWMutex
+	count int
+}
+
+func (mc *msgCount) Inc() {
+	mc.lock.Lock()
+	defer mc.lock.Unlock()
+	mc.count++
+}
+
+func (mc *msgCount) Count() int {
+	mc.lock.RLock()
+	defer mc.lock.RUnlock()
+	return mc.count
+}
```

Reply:
Great suggestion. I've updated the PR.

Reply:
Clearly, I cannot use GitHub's review feature. :dagger:  I'm still reviewing but all these comments went out.

*Just a fun note* here perf does not matter. I updated the patch later to use atomic but you were super fast. :)

atomics is faster than lock for this kind of incremental counter
```
Benchmark_incAtomic-4   	       2	 825926497 ns/op
Benchmark_incLock-4     	       1	1558614020 ns/op
```
https://gist.github.com/frncmx/0cef773d28609120d0493a72de2956a0

Reply:
Yes, atomics are cool for improving bottlenecks.